src/go/rdctl/pkg/reg/reg.go

```diff
@@ -0,0 +1,160 @@
+		return []string{fmt.Sprintf(`"%s"=dword:%d`, jsonTag, boolValue)}, false, nil
+	case reflect.Int, reflect.Int8, reflect.Int16,
+		reflect.Int32, reflect.Uint, reflect.Uint8, reflect.Uint16,
+		reflect.Uint32:
+		return []string{fmt.Sprintf(`"%s"=dword:%x`, jsonTag, v.Int())}, false, nil
+	case reflect.Int64, reflect.Uint64:
+		return []string{fmt.Sprintf(`"%s"=qword:%x`, jsonTag, v.Int())}, false, nil
+	case reflect.Float32, reflect.Float64:
+		return []string{fmt.Sprintf(`"%s"=dword:%x`, jsonTag, int(v.Float()))}, false, nil
+	case reflect.String:
+		return []string{fmt.Sprintf(`"%s"="%s"`, jsonTag, escape(v.String()))}, false, nil
+	}
+	return nil, true, fmt.Errorf("convertToRegFormat: don't know how to process kind: %s, (%T), value: %v for var %s\n", kind, v, v, jsonTag)
+}
+
+// Encode multi-stringSZ settings in comma-separated ucs2 little-endian bytes
+// e.g.=> ["abc", "def"] would be ucs-2-encoded as '61,00,62,00,63,00,00,00,64,00,65,00,66,00,00,00,00,00'
+// where a null 16-bit word (so two 00 bytes) separate each pair of words and
+// two null 16-bit words ("00 00 00 00") indicate the end of the list
+func stringToMultiStringHexBytes(values []string) string {
+	valueString := strings.Join(values, "\x00")
+	hex := utf16.Encode([]rune(valueString))
+	hexChars := make([]string, len(hex)*2)
+	for i, h := range hex {
+		s := fmt.Sprintf("%04x", h)
+		hexChars[2*i] = s[2:4]
+		hexChars[2*i+1] = s[0:2]
+	}
+	return strings.Join(hexChars, ",") + ",00,00,00,00"
+}
+
+// JsonToReg - convert the json settings to a reg file
+/**
+ * @param hiveType: "hklm" or "hkcu"
+ * @param profileType: "defaults" or "locked"
+ * @param settingsBodyAsJSON - options marshaled as JSON
+ * @returns: array of strings, intended for writing to a reg file
+ */
+func JsonToReg(hiveType string, profileType string, settingsBodyAsJSON string) ([]string, error) {
+	var settings options.ServerSettingsForJSON
+
+	fullHiveType, ok := map[string]string{"hklm": "HKEY_LOCAL_MACHINE", "hkcu": "HKEY_CURRENT_USER"}[hiveType]
+	if !ok {
+		return nil, fmt.Errorf("unrecognized hiveType of '%s', must be 'hklm' or 'hkcu'", hiveType)
+	}
+	_, ok = map[string]bool{"defaults": true, "locked": true}[profileType]
+	if !ok {
+		return nil, fmt.Errorf("unrecognized profileType of '%s', must be 'defaults' or 'locked'", profileType)
+	}
+	if err := json.Unmarshal([]byte(settingsBodyAsJSON), &settings); err != nil {
+		return nil, fmt.Errorf("error in json: %s\n", err)
+	}
+	headerLines := []string{"Windows Registry Editor Version 5.00"}
+	bodyLines, _, err := convertToRegFormat([]string{fullHiveType, "SOFTWARE", "Policies", "Rancher Desktop", profileType}, reflect.ValueOf(settings), "")
+	if err != nil {
+		return nil, err
+	}
+	if len(bodyLines) > 0 {
+		headerLines = append(headerLines, "", fmt.Sprintf("[%s\\%s\\%s]", fullHiveType, "SOFTWARE", "Policies"))
+		headerLines = append(headerLines, "", fmt.Sprintf("[%s\\%s\\%s\\%s]", fullHiveType, "SOFTWARE", "Policies", "Rancher Desktop"))
```

Comment:
I went with the empty lines because that's what exporting from the registry gave me, but you're right, they aren't needed.

And I thought it might be useful to keep them for round-trip testing, but the export routine writes its output in `ucs2` with a BOM, and it writes out DWORDS with 8 characters, while we don't bother with the leading zeroes. So I can drop them.